hazelcast/src/main/java/com/hazelcast/query/impl/getters/Extractors.java

```diff
@@ -131,14 +132,17 @@ private Getter instantiateGetter(Object targetObject, String attributeName) {
         } else {
             if (targetObject instanceof Data) {
                 if (((Data) targetObject).isPortable()) {
-                    //targetObject may be a Data whose object form is a json?
                     if (genericPortableGetter == null) {
                         // will be initialised a couple of times in the worst case
                         genericPortableGetter = new PortableGetter(ss);
                     }
                     return genericPortableGetter;
                 } else if (((Data) targetObject).isJson()) {
-                    return JsonDataGetter.INSTANCE;
```

Comment:
why not a static instance?

Reply:
JsonDataGetter requires serialization service in a followup PR. That's why I left it like this here.